src/core/src/core_logic/PatchInstaller.py

```diff
@@ -98,16 +98,16 @@ def start_installation(self, simulate=False):
 
         return overall_patch_installation_successful
 
-    def raise_if_agent_incompatible(self):
+    def raise_if_telemetry_unsupported(self):
         if self.lifecycle_manager.get_vm_cloud_type() == Constants.VMCloudType.ARC and self.execution_config.operation not in [Constants.ASSESSMENT, Constants.INSTALLATION]:
-            self.composite_logger.log("Skipping agent compatibility check for Arc cloud type when operation is not manual")
+            self.composite_logger.log("Skipping telemetry compatibility check for Arc cloud type when operation is not manual")
```

Comment:
It's impossible to hit these lines because assessment always runs before installation and also has this check, so it will fail the entire operation before getting to the installer. I added one test to mock a no-op assessment so that these lines can be covered by tests, but I had a question: How can this line be hit if the operation isn't installation? CoreMain specifically sets the operation to installation before starting the patch installer, so this should (?) always fail.